src/Integration/Notifications/NotificationIndicatorViewModel.cs

```diff
@@ -0,0 +1,175 @@
+
+        public ObservableCollection<SonarQubeNotification> NotificationEvents { get; }
+
+        // For testing
+        public NotificationIndicatorViewModel()
+            : this (ThreadHelper.Generic.Invoke)
+        {
+        }
+
+        public NotificationIndicatorViewModel(Action<Action> uiThreadInvoker)
+        {
+            this.uiThreadInvoker = uiThreadInvoker;
+
+            NotificationEvents = new ObservableCollection<SonarQubeNotification>();
+            text = GetTooltipText();
+
+            autocloseTimer = new TimerWrapper { Interval = 3000, AutoReset = false };
+            autocloseTimer.Elapsed += OnAutocloseTimerElapsed;
+
+            ClearUnreadEventsCommand = new RelayCommand(() => HasUnreadEvents = false);
+        }
+
+        public ICommand ClearUnreadEventsCommand { get; }
+
+        public string TooltipText
+        {
+            get
+            {
+                return text;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref text, value);
+            }
+        }
+
+        public bool HasUnreadEvents
+        {
+            get
+            {
+                return hasUnreadEvents;
+            }
+
+            set
+            {
+                SetAndRaisePropertyChanged(ref hasUnreadEvents, value);
+                TooltipText = GetTooltipText();
+            }
+        }
+
+        public bool IsIconVisible
+        {
+            get
+            {
+                return isVisible;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref isVisible, value);
+            }
+        }
+
+        public bool AreNotificationsEnabled
+        {
+            get
+            {
+                return areNotificationsEnabled;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref areNotificationsEnabled, value);
+            }
+        }
+
+        public bool IsBalloonTooltipVisible
+        {
+            get
+            {
+                return isBalloonTooltipVisible;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref isBalloonTooltipVisible, value);
+
+                // If the tooltip was closed manually, stop the timer
+                if (!isBalloonTooltipVisible)
+                {
+                    autocloseTimer.Stop();
+                }
+            }
+        }
+
+        public void SetNotificationEvents(IList<SonarQubeNotification> events)
+        {
+            if (events == null ||
+                events.Count == 0 ||
+                !AreNotificationsEnabled ||
+                !isVisible)
+            {
+                return;
+            }
+
+            uiThreadInvoker(() =>
+                {
+                    NotificationEvents.Clear();
+
+                    foreach (var ev in events)
+                    {
+                        NotificationEvents.Add(ev);
+                    }
```

Comment:
We cannot really do better with the fwk we are using but I really don't like this pattern. `Clear` will trigger a refresh of the UI and all add will do the same. So you end up triggering a lot of refresh while the only one you care about is the last one..... If only we were using nice UI framework (https://reactiveui.net/docs/handbook/collections/reactive-list#suppressing-notifications).